var/spack/repos/builtin/packages/chai/package.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Chai(CMakePackage, CudaPackage):
+    """
+    Copy-hiding array interface for data migration between memory spaces
+    """
+
+    homepage = "https://github.com/LLNL/CHAI"
+    git      = "https://github.com/LLNL/CHAI.git"
+
+    version('develop', branch='develop', submodules='True')
+    version('main', branch='main', submodules='True')
+    version('2.1.1', tag='v2.1.1', submodules='True')
+    version('2.1.0', tag='v2.1.0', submodules='True')
+    version('2.0.0', tag='v2.0.0', submodules='True')
+    version('1.2.0', tag='v1.2.0', submodules='True')
+    version('1.1.0', tag='v1.1.0', submodules='True')
+    version('1.0', tag='v1.0', submodules='True')
+
+    depends_on('cmake@3.8:', type='build')
```

Comment:
Do we need a variant for CUDA support?  How will the user know that they need to select both +cuda and +cuda_arch ?

Reply:
Cuda variant is added by heritage of CudaPackage class.

Reply:
Cool!  I didn't know that existed!  Does that pull in cmake package options as well?

Reply:
This is done by double inheritance:
https://github.com/spack/spack/blob/ce6126d8ce4bc40f688932c0df9385fb3c57582c/var/spack/repos/builtin/packages/chai/package.py#L9